crates/esthri-cli/src/cli_utils.rs

```diff
@@ -136,3 +138,67 @@ pub fn setup_upload_termination_handler(s3: S3Client) {
         std::process::exit(0);
     });
 }
+
+pub fn setup_s3client_with_cred_provider() -> Result<S3Client> {
+    let mut hyper_builder = Client::builder();
+    hyper_builder.pool_idle_timeout(Duration::from_secs(20));
+
+    let https_connector = esthri::new_https_connector();
+    let http_client = HttpClient::from_builder(hyper_builder, https_connector);
+
+    match std::env::var("ESTHRI_CREDENTIAL_PROVIDER") {
+        Ok(val) => match val.as_str() {
+            "env" => {
+                let credentials_provider = EnvironmentProvider::default();
+                Ok(S3Client::new_with(
+                    http_client,
+                    credentials_provider,
+                    Region::default(),
+                ))
+            }
+            "profile" => {
+                let credentials_provider = ProfileProvider::new().unwrap();
+                Ok(S3Client::new_with(
+                    http_client,
+                    credentials_provider,
+                    Region::default(),
+                ))
+            }
+            "container" => {
+                let credentials_provider = ContainerProvider::new();
+                Ok(S3Client::new_with(
+                    http_client,
+                    credentials_provider,
+                    Region::default(),
+                ))
+            }
+            "instance_metadata" => {
+                let credentials_provider = InstanceMetadataProvider::new();
+                Ok(S3Client::new_with(
+                    http_client,
+                    credentials_provider,
+                    Region::default(),
+                ))
+            }
+            "esthri" => {
```

Comment:
```suggestion
            "" => {
```

Is there any downside to making empty and unset do the same thing?  It's a pretty common convention in environment variable handling to do this (a convention that we've used in other Swift tools too).

Reply:
@silverjam No downside that I can see, happy to align with other tool's convention

Reply:
👍